src/Core/Plugin.php

```diff
@@ -170,6 +184,24 @@ public static function load($plugin, array $config = [])
     }
 
     /**
+     * Loads the config mappings for plugins.
+     *
+     * The map data in CONFIG/plugins.php map data is used set default paths
+     * for where plugins are located.
+     *
+     * @return void
+     */
+    protected static function _loadConfigMap()
+    {
+        if (!file_exists(CONFIG . 'plugins.php')) {
+            static::$_config = [];
+            return;
+        }
+        $config = include CONFIG . 'plugins.php';
```

Comment:
Instead of using `include` how about using `Configure::load()` and also adding a methods to update and dump the map?  That way all the code for maintaining the map remains in core and the plugin-installer would just need to call the appropriate method upon plugin installation / removal.